server/bitbucketclient/factory.go

```diff
@@ -0,0 +1,24 @@
+package bitbucketclient
+
+import (
+	"fmt"
+	"net/http"
+
+	bitbucketv1 "github.com/gfleury/go-bitbucket-v1"
+)
+
+type ClientConfiguration struct {
+	SelfHostedURL    string
+	SelfHostedAPIURL string
```

Comment:
I've been thinking that this `SelfHostedAPIURL` is derived from `SelfHostedURL` (we just add some suffix to it), so maybe we can have just one config setting `SelfHostedURL`, and derive `SelfHostedAPIURL` at runtime?